### PR TITLE
De-duplicate doc index entries

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

File: documentation/source/library_reference.rst [hunk 1]
```diff
@@ -139,6 +139,7 @@ Clock
 -----
 
 .. autoclass:: cocotb.clock.Clock
+    :noindex:
```
Comment (eric-wieser) [inline documentation/source/library_reference.rst:142]: Where's the other reference? A comment might be good here

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Done.

File: documentation/source/library_reference.rst [hunk 2]
```diff
@@ -141,6 +141,9 @@ Clock
 .. autoclass:: cocotb.clock.Clock
     :members:
     :member-order: bysource
+    :noindex:
+..
+   Primary occurrence in this document under "Interacting with the Simulator", so don't index this
```
Comment (eric-wieser) [inline documentation/source/library_reference.rst:146]: Might make more sense to remove one of these, documenting the same class twice on the same page seems confusing.

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Done.

**Pull request merged**